alicloud/data_source_alicloud_ess_alarms_test.go

```diff
@@ -0,0 +1,175 @@
+			"alarms.0.name":                   "tf-testAccDataSourceEssAlarms",
+			"alarms.0.scaling_group_id":       CHECKSET,
+			"alarms.0.description":            "",
+			"alarms.0.metric_type":            "system",
+			"alarms.0.metric_name":            CHECKSET,
+			"alarms.0.period":                 CHECKSET,
+			"alarms.0.statistics":             CHECKSET,
+			"alarms.0.threshold":              CHECKSET,
+			"alarms.0.comparison_operator":    CHECKSET,
+			"alarms.0.evaluation_count":       CHECKSET,
+			"alarms.0.cloud_monitor_group_id": CHECKSET,
+		}
+	}
+
+	var fakeEssAlarmsMapFunc = func(rand int) map[string]string {
+		return map[string]string{
+			"alarms.#": "0",
+			"ids.#":    "0",
+			"names.#":  "0",
+		}
+	}
+
+	var essAlarmsCheckInfo = dataSourceAttr{
+		resourceId:   "data.alicloud_ess_alarms.default",
+		existMapFunc: existEssAlarmsMapFunc,
+		fakeMapFunc:  fakeEssAlarmsMapFunc,
+	}
+
+	essAlarmsCheckInfo.dataSourceTestCheck(t, -1, scalingGroupIdConf, nameRegexConf, idsConf, allConf)
+}
+
+func testAccCheckAlicloudEssAlarmsDataSourceConfig(rand int, attrMap map[string]string) string {
+	var pairs []string
+	for k, v := range attrMap {
+		pairs = append(pairs, k+" = "+v)
+	}
+
+	config := fmt.Sprintf(`
+variable "name" {
+	default = "tf-testAccDataSourceEssAlarms"
+}
+
+data "alicloud_zones" "default" {
+  available_disk_category     = "cloud_efficiency"
+  available_resource_creation = "VSwitch"
+}
+
+data "alicloud_instance_types" "default" {
+  availability_zone = "${data.alicloud_zones.default.zones.0.id}"
+  cpu_core_count    = 2
+  memory_size       = 4
+}
+
+data "alicloud_images" "default" {
+  name_regex  = "^ubuntu_18.*64"
+  most_recent = true
+  owners      = "system"
+}
+
+resource "alicloud_vpc" "default" {
+  name       = "${var.name}"
+  cidr_block = "172.16.0.0/16"
+}
+
+resource "alicloud_vswitch" "default" {
+  vpc_id            = "${alicloud_vpc.default.id}"
+  cidr_block        = "172.16.0.0/24"
+  availability_zone = "${data.alicloud_zones.default.zones.0.id}"
+  name              = "${var.name}"
+}
+
+resource "alicloud_security_group" "default" {
+  name   = "${var.name}"
+  vpc_id = "${alicloud_vpc.default.id}"
+}
+
+resource "alicloud_security_group_rule" "default" {
+  	type = "ingress"
+  	ip_protocol = "tcp"
+  	nic_type = "intranet"
+  	policy = "accept"
+  	port_range = "22/22"
+  	priority = 1
+  	security_group_id = "${alicloud_security_group.default.id}"
+  	cidr_ip = "172.16.0.0/24"
+}
+
+resource "alicloud_ess_scaling_group" "default" {
+	min_size = 0
+	max_size = 2
+	default_cooldown = 20
+	removal_policies = ["OldestInstance", "NewestInstance"]
+	scaling_group_name = var.name
+	vswitch_ids = ["${alicloud_vswitch.default.id}"]
+}
+
+resource "alicloud_ess_scaling_rule" "default"{
+	scaling_group_id = "${alicloud_ess_scaling_group.default.id}"
+	scaling_rule_name = var.name
+	adjustment_type = "QuantityChangeInCapacity"
+	adjustment_value = 1
+}
+resource "alicloud_ess_alarm" "default"{
```

Comment:
Please remove the redundant resources and only retain which is related to `alarm`.